src/components/shared/GitHubLibrary/components/AddGitHubLibraryDialogContent.tsx

```diff
@@ -0,0 +1,241 @@
+  action: GitHubLibraryFormAction,
+): GitHubLibraryProcessState {
+  switch (action.type) {
+    case "UPDATE": {
+      const newState = { ...state, ...action.payload };
+      return {
+        ...newState,
+        hasErrors: !isStateValid(newState),
+      };
+    }
+    default:
+      return state;
+  }
+}
+
+function ConfigureImport({
+  onSuccess,
+}: {
+  onSuccess: (components: HydratedComponentReference[]) => void;
+}) {
+  const notify = useToastNotification();
+
+  const initialState: GitHubLibraryProcessState = {
+    repoName: "",
+    accessToken: "",
+    hasErrors: true,
+  };
+
+  const [state, dispatch] = useReducer(addGithubLibraryReducer, initialState);
+
+  const { mutate: importComponentsFromGitHubLibrary, isPending } = useMutation({
+    mutationFn: async (state: GitHubLibraryProcessState) => {
+      const files = await fetchGitHubFiles(
+        state.accessToken,
+        state.repoName,
+        (name) => name.endsWith(".component.yaml"),
+      );
+
+      const hydratedComponents = await hydrateAllComponents(
+        files.map((file) => ({
+          url: file.url,
+          text: file.content,
+        })),
+      );
+
+      // create a library with the hydrated components
+      await ensureGithubLibrary({
+        repoName: state.repoName,
+        accessToken: state.accessToken,
+        components: hydratedComponents,
+      });
+
+      return hydratedComponents;
+    },
+    onSuccess: (components) => {
+      notify(`Successfully fetched ${components.length} components`, "success");
+      onSuccess(components);
+    },
+    onError: (error) => {
+      notify(
+        `Error importing components from GitHub library: ${error.message}`,
+        "error",
+      );
+    },
+  });
+
+  const handleSubmit = useCallback(async () => {
+    if (state.hasErrors) {
+      notify("Please fill in all fields", "error");
+      return;
+    }
+
+    await importComponentsFromGitHubLibrary(state);
+  }, [importComponentsFromGitHubLibrary, state, notify]);
+
+  return (
+    <BlockStack
+      gap="4"
+      className={cn(isPending && "pointer-events-none opacity-50")}
+    >
+      <Paragraph tone="subdued" size="xs">
+        Connect a GitHub repository to import components from your library. You
+        can use a personal access token to access repositories.
+      </Paragraph>
+      <InputField
+        id="repo-name"
+        label="Repository Name"
+        placeholder="owner/repository"
+        value={state.repoName}
+        validate={validateRepoName}
+        onChange={(value) => {
+          dispatch({ type: "UPDATE", payload: { repoName: value ?? "" } });
+        }}
+      />
+      <InputField
+        id="pat"
+        label="Personal Access Token"
+        placeholder="ghp_..."
+        value={state.accessToken}
+        validate={validatePAT}
+        onChange={(value) => {
+          dispatch({ type: "UPDATE", payload: { accessToken: value ?? "" } });
+        }}
```

Comment:
I'm thinking out loud here:

```suggestion
        onChange={(value) => {
          if (value) {
              dispatch({ type: "UPDATE", payload: { accessToken: value } });
          }
        }}
```

(something about `value ?? ""` bothers me here and elsewhere in the app)